sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -616,6 +616,17 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   def as(alias: String): Column = Alias(expr, alias)()
 
   /**
+   * Assigns the given aliases to the results of a table generating function.
+   * {{{
+   *   // Renames colA to colB in select output.
+   *   df.select(explode($"myMap").as("key" :: "value" :: Nil))
+   * }}}
+   *
+   * @group expr_ops
+   */
+  def as(aliases: Seq[String]): Column = MultiAlias(expr, aliases)
```

Comment:
MultiAlias can be used for any table generating function (and others can already be called using callUDF) so we will have to defer the check to the analyzer (where we already do it).